torchdata/datapipes/iter/util/distributed.py

```diff
@@ -171,31 +171,34 @@ def _callback_fn(self, exp: Expected) -> None:
 
     def __iter__(self) -> Iterator[T_co]:
         assert self._executor is None
-
         if not (dist.is_available() and dist.is_initialized()):
-            raise RuntimeError("Torch Distributed is required to be initialized")
+            raise RuntimeError("Torch Distributed is required to be initialized to use `FullSync`.")
+
         if self._process_group is None:
             self._process_group = dist.new_group(backend="gloo")
         self._world_size = dist.get_world_size()
 
-        self._executor = _PrefetchExecutor(iter(self.datapipe), 1, self._callback_fn, self.timeout)
-        while True:
-            with self._cv:
-                is_success = self._cv.wait_for(
-                    lambda: self._done_callback is True,
-                    self.timeout,
-                )
-                if not is_success:
-                    raise PrefetchTimeoutError(self.timeout)
-                if self._error is not None:
-                    raise self._error
-                if bool(self._sync_counter < self._world_size):
+        if self._world_size == 1:  # The below functionalities are not needed if `_world_size == 1`
+            yield from self.datapipe
```

Comment:
Nit comment and let you decide if you want to do it.
you can `return` after `yield from self.datapipe` within the clause of `if self._world_size == 1:` to reduce the `else` clause.